core/vm/zk_batch_counters.go

```diff
@@ -166,6 +166,39 @@ func (bcc *BatchCounterCollector) CheckForOverflow(verifyMerkleProof bool) (bool
 	return overflow, nil
 }
 
+// OverflowCounterStats returns a string with combined counter stats if any counter has overflowed.
+// If no overflow is detected, it returns an error. If a Go error occurs, it also returns that error.
+func (bcc *BatchCounterCollector) OverflowCounterStats(verifyMerkleProof bool) (string, error) {
+	combined, err := bcc.CombineCollectors(verifyMerkleProof)
+	if err != nil {
+		return "", err
+	}
+
+	var overflow bool
+	var logText string
+
+	// Check if any counter has overflowed
+	for _, v := range combined {
+		if v.remaining < 0 {
+			overflow = true
+			break
+		}
+	}
+
+	// If no overflow, return an error
+	if !overflow {
+		return "", fmt.Errorf("no counter overflow detected")
```

Comment:
Returning an error in the case of no overflow doesn't seem intuitive.  No overflows is a good thing.

Reply:
okay, gonna remove it!